validator/tests/test_report.py

```diff
@@ -1,37 +1,31 @@
 """Tests for delphi_validator.report."""
-from datetime import date
-from delphi_validator.errors import ValidationError
+from delphi_validator.errors import ValidationFailure
 from delphi_validator.report import ValidationReport
 
 class TestValidationReport:
     """Tests for ValidationReport class."""
 
-    ERROR_1 = ValidationError(("good", date(2020, 10, 5)), "exp 1", "msg 1")
-    ERROR_2 = ValidationError(("bad", date(2020, 11, 18)), "exp 2", "msg 2")
+    ERROR_1 = ValidationFailure("good", "data 1", "msg 1")
+    ERROR_2 = ValidationFailure("bad", "data 2", "msg 2")
 
     def test_add_raised_unsuppressed_error(self):
         """Test that an unsupressed error shows up in the unsuppressed error list."""
-        report = ValidationReport([("bad", "2020-10-05")])
+        report = ValidationReport(set([("bad", "data 1")]))
         report.add_raised_error(self.ERROR_1)
         report.add_raised_error(self.ERROR_2)
         assert report.unsuppressed_errors == [self.ERROR_1, self.ERROR_2]
 
     def test_add_raised_suppressed_error(self):
         """Test that an supressed error does not show up in the unsuppressed error list."""
-        report = ValidationReport([("good", "2020-10-05")])
+        report = ValidationReport([("good", "data 1")])
         report.add_raised_error(self.ERROR_1)
 
         assert len(report.unsuppressed_errors) == 0
         assert report.num_suppressed == 1
-        assert len(report.errors_to_suppress) == 0
-
-        # Each error can only be surpressed once.
-        report.add_raised_error(self.ERROR_1)
-        assert report.unsuppressed_errors == [self.ERROR_1]
 
     def test_str(self):
         """Test that the string representation contains all information."""
-        report = ValidationReport([("good", "2020-10-05")])
+        report = ValidationReport(set([("good", "data 1")]))
```

Comment:
I think this is a set which contains a list, which will error out because a list isn't hashable. I think you either want `set(("good", "data 1"))` or `set(tuple(["good", "data 1"]))`.

Reply:
This syntax constructs a set from the elements of the list and works as is.